### ISSUE TITLE
Need to make a Custom View for tab bar

### ISSUE THREAD

**Issue by obadasemary** [rechsteiner/Parchment#256]
Hi I need to make a custom view for tab bar like the image 

<img width="311" alt="screen shot 2018-09-28 at 9 47 11 pm" src="https://user-images.githubusercontent.com/7041989/46227645-83336a00-c368-11e8-8341-a0fd322c1307.png">

How Can I make it

**Comment by rechsteiner** [comment]
Hi @obadasemary!

It depends a bit how dynamic you need it to be 🙂

**Option 1**:

If it's always three items then you can create your own `PagingTitleCell` where you apply corner radius based on the index of the paging item. E.g:

```Swift
class RoundedPagingCell: PagingTitleCell {

   open override func setPagingItem(_ pagingItem: PagingItem, selected: Bool, options: PagingOptions) {
    super.setPagingItem(pagingItem, selected: selected, options: options)
    guard let item = pagingItem as? PagingIndexItem { return }
    
    // Apply shadow to content view

    if item.index == 0 {
        // Apply left corner radius to contentView
    } else if item.index == 2 {
        // Apply right corner radius to contentView
    }
    
  }
    
}
```
**Option 2**:

If you need the number of items to be dynamic, I think I would create my own `PagingTitleItem` that contains a title and an enum saying how it should be rounded. E.g. something like this:

```Swift
struct RoundedPagingItem: PagingTitleItem {
    enum Rounding {
        case leading
        case middle
        case trailing
    }

    let title: String
    let rounding: Rounding
}
```

Then create your own `PagingTitleCell ` subclass where you round the corners of the cell based on the `rounding` property on you `PagingItem`:

```Swift
class RoundedPagingCell: PagingTitleCell {

   open override func setPagingItem(_ pagingItem: PagingItem, selected: Bool, options: PagingOptions) {
    super.setPagingItem(pagingItem, selected: selected, options: options)
    guard let item = pagingItem as? CustomPagingItem { return }
    
    switch item.rounding {
        case .leading:
              // Apply left corner radius
        case .trailing:
              // Apply left corner radous
        case .none:
             break
      }
  }
    
}
```

Then setup your [custom datasource](https://github.com/rechsteiner/Parchment/blob/master/Documentation/data-source.md) with some data like this:

```Swift
let data = [
      RoundedPagingItem(title: "First", rounding: .leading),
      RoundedPagingItem(title: "Second", rounding: .none),
      RoundedPagingItem(title: "Third", rounding: .none),
      RoundedPagingItem(title: "Fourth", rounding: .trailing)
]
```

Let me know if that works for you!

**Comment by rechsteiner** [comment]
You can apply corner radius to the `contentView` property in your `PagingTitleCell` subclass

**Comment by swordark** [comment]
@rechsteiner  I applied it directly to PagingTitleCell self. Now i have better result. (Because i am using selected item background and it is changing whole background of item, not just content views background) Thank you. 

Also i am trying to add shadow like the image above. But when i add shadow, i couldnt make it visible at the top and bottom of menu item cell. I used menuInsets but still shadow is not visible at bottom and top.
